### PR TITLE
Feat/level editor

### PR DESCRIPTION
**Pull request by NateCastleOIT**
Added Editor Grid and Grid snapping.

### CODE REVIEW

File: TKGame/TKGame.cs [hunk 1]
```diff
@@ -110,6 +110,10 @@ protected override void LoadContent()
             EntityManager.Add(Enemy.Instance);
             EntityManager.Add(Item.Instance);
 
+            EntityManager.Add(Enemy.Instance);
+
+            EntityManager.Add(Enemy.Instance);
+
```
Comment (aaronw84) [inline TKGame/TKGame.cs:116]: Not sure if this is what you intended but calling to add an enemy instance multiple times will just make the enemy update() be executed more frequently.

**Review by aaronw84 — APPROVED** [review]

File: TKGame/Level Editor Content/Stages/defaultStage.json [hunk 2]
```diff
@@ -2,8 +2,8 @@
   {
```
Comment (NathanGr33n) [inline TKGame/Level Editor Content/Stages/defaultStage.json:2]: We will need some kind of map of levels in the future possibly. Looks good for now though.

**Review by NathanGr33n — APPROVED** [review]
Looks good, Might need some updates further on to deal with new features.

File: TKGame/TKGame.cs (hunk 1 above)
Comment (NateCastleOIT) [reply TKGame/TKGame.cs:116]: I was having some weird stashed change issues, will be merged out

**Pull request merged**